cpp/CMakeLists.txt

```diff
@@ -840,3 +841,79 @@ add_custom_command(OUTPUT CUDF_DOXYGEN
                    VERBATIM)
 
 add_custom_target(docs_cudf DEPENDS CUDF_DOXYGEN)
+
+
+####################################################################################################
+# - cuDatasource - OFF by default due to increased number of dependencies
+if(BUILD_CUDATASOURCE)
+
+    # cuDatasource library
+    add_library(libcudatasource
+        cudatasource/kafka/src/kafka_consumer.cpp
+    )
+
+    # Include paths
+    include_directories("${CMAKE_SOURCE_DIR}/cudatasource/kafka/include")
+    
+    # Rename installation to proper name for later finding
+    set_target_properties(libcudatasource PROPERTIES OUTPUT_NAME "cuDatasource")
+    set_target_properties(libcudatasource PROPERTIES BUILD_RPATH "\$ORIGIN")
+
+    ###################################################################################################
+    # cuDatasource - Kafka ------------------------------------------------------------------------------------
+
+    find_path(RDKAFKA_INCLUDE "librdkafka" HINTS "$ENV{RDKAFKA_ROOT}/include")
+    find_library(RDKAFKA++_LIBRARY "rdkafka++" HINTS "$ENV{RDKAFKA_ROOT}/lib" "$ENV{RDKAFKA_ROOT}/build")
+
+    message(STATUS "RDKAFKA: RDKAFKA++_LIBRARY set to ${RDKAFKA++_LIBRARY}")
+    message(STATUS "RDKAFKA: RDKAFKA_INCLUDE set to ${RDKAFKA_INCLUDE}")
+
+    target_link_libraries(libcudatasource ${RDKAFKA++_LIBRARY})
+
+    ###################################################################################################
+    # - cuDatasource Install ----------------------------------------------------------------------------------
+    target_link_libraries(cudf libcudatasource)
+
+    install(TARGETS libcudatasource
+            DESTINATION lib
+            COMPONENT libcudatasource)
+
+    install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/cudatasource/kafka/include
+            DESTINATION include
+            COMPONENT libcudatasource)
+
+    add_custom_target(install_cudatasource
+            COMMAND "${CMAKE_COMMAND}" -DCOMPONENT=libcudatsource -P "${CMAKE_BINARY_DIR}/cmake_install.cmake"
+            DEPENDS libcudatasource)
+
+    ####################################################################################################
+    # - cuDatasource Tests
+    if(BUILD_TESTS)
+        if(GTEST_FOUND)
+            message(STATUS "Google C++ Testing Framework (Google Test) found in ${GTEST_ROOT}")
+            include_directories(${GTEST_INCLUDE_DIR})
+            add_subdirectory(${CMAKE_SOURCE_DIR}/cudatasource/kafka/tests)
+        else()
+            message(AUTHOR_WARNING "Google C++ Testing Framework (Google Test) not found: automated tests are disabled.")
+        endif(GTEST_FOUND)
+    endif(BUILD_TESTS)
+        
+    message(STATUS "CUDATASOURCE_TEST_LIST set to: ${CUDATASOURCE_TEST_LIST}")
+
+    add_custom_target(build_tests_cudatasource
+                      DEPENDS ${CUDATASOURCE_TEST_LIST})
+
+    add_custom_target(test_cudatasource
+                      COMMAND ctest -E "CUDATASOURCE"
```

Comment:
This excludes the tests as opposed to including them.

Reply:
copy pasta. Will remove `-E` flag